### PR TITLE
Gracefully handle long time intervals

### PR DESCRIPTION
**Pull request by DPeterK**
There's an odd discrepancy between `UDUNITS2`, which will handle months and years as valid time intervals, and `netcdftime`, which won't. This means you can produce a valid time unit with a time interval of years, which you then cannot convert between dates and numbers. 

This PR adds a new method `is_long_time_interval` to the `Unit` class -- does this unit instance describe a time unit with a long interval (that is, months or years)? If it does and it is used in a date <--> number conversion then the unit will raise an exception, rather than the error being raised in the depths of `netcdftime`.

Downstream, the above discrepancy affects Iris coord printing. If the `__str__` method of a coord is called and the coord is a time coord, then the `__str__` will try and produce pretty time coord using `num2date`, which will error, causing the somewhat horrid situation of a print statement causing an exception.

### CODE REVIEW

**Comment by DPeterK** [comment]
Note that as part of this change I've modified how `cf_units.num2date` and `cf_units.date2num` function. They now create a `cf_units.Unit` instance and use that instance to run the conversion so that the new long time interval exception in cf_units is raised and not the exception in `netcdftime`.

File: cf_units/__init__.py [hunk 1]
```diff
@@ -1210,6 +1209,35 @@ def is_time_reference(self):
         """
         return self.calendar is not None
 
+    def is_long_time_interval(self):
+        """
+        Defines whether this unit describes a time unit with a long time
+        interval ("months" or "years"). These long time intervals *are*
+        supported by `UDUNITS2` by are not supported by `netcdftime`. This
```
Comment (lbdreyer) [inline cf_units/__init__.py]: typo: These long time intervals are supported by `UDUNITS2` **but** are not supported by `netcdftime`.

**Comment by DPeterK** [comment]
Thanks @lbdreyer - I've corrected the typo. 

Any more for any more?

**Comment by DPeterK** [comment]
And now the license headers tests fail...

**Comment by lbdreyer** [comment]
I'm happy with the implementation + tests. Thanks @dkillick !

**Comment by pelson** [comment]
@dkillick - sorry to drag this back up, but did you raise this with the netcdftime repository? Seems like a thing they may be interested in addressing there...

**Comment by DPeterK** [comment]
@pelson I did not - I should. We should also try and get the associated Iris PR https://github.com/SciTools/iris/pull/2354 merged, a job which is a bit overdue...

Edit: note that merging the above-linked _still_ requires a release of cf_units.

**Comment by pelson** [comment]
> Edit: note that merging the above-linked still requires a release of cf_units.

cf_units 1.2.0 has the magic. 👍

**Pull request merged**